File: design-patterns/go/composite/player_group/basketball_player.go
// baseketball_player.go

package main

import "fmt"

type BasketBallPlayer struct {
	name  string
	age   int
	point int
}

func NewBasketBallPlayer(name string, age int, point int) (basketballPlayer *BasketBallPlayer) {
	basketballPlayer = &BasketBallPlayer{}
	basketballPlayer.name = name
	basketballPlayer.age = age
	basketballPlayer.point = point
	return
}

func (basketballPlayer *BasketBallPlayer) PrintDetails() {
	fmt.Println("Game: Basketball")
	fmt.Printf("Name: %v\n", basketballPlayer.name)
	fmt.Printf("Age: %v\n", basketballPlayer.age)
	fmt.Printf("Points: %v\n", basketballPlayer.point)
}

File: design-patterns/php/composite/playergroup/BasketballPlayer.php
<?php
// BasketballPlayer.php

namespace BigBoxCode\DesignPattern\Composite\PlayerGroup;


class BasketballPlayer implements Player {

    public function __construct(
        private string $name,
        private int $age,
        private int $point
    ) {

    }

    public function printDetails(): void {
        echo "\nGame: Basketball";
        echo "\nName: " . $this->name;
        echo "\nAge: " . $this->age;
        echo "\nPoints: " . $this->point;
    }
}